folly/io/async/AsyncPipe.cpp

```diff
@@ -48,8 +48,6 @@ void AsyncPipeReader::close() {
 
     if (closeCb_) {
       closeCb_(fd_);
-    } else {
-      netops::close(fd_);
```

Comment:
Removing this seems wrong - close() no longer closes the underlying fd when closeCb_ is not defined?

Reply:
This fd was added in async list by underlying openssl. Isn't it better to let the ssl library close the fd once operation is complete. 

Also, If I close here, my driver handle will be closed and cannot be used for any further in the SSL connection.

Reply:
@afrind  I will elaborate my use case too here:

I have a crypto device to which asymmetric and symmetric operations from openssl are offloaded. Instead of pipe fd, I am using the driver handle of this device as an async fd, which is signalled whenever the crypto operation is completed.

So this handle has to be open throughout the length of an SSL connection. Also in this design,  for each of crypto  operation done in ASYNC mode, the SSL_ERROR_WANT_ASYNC is returned. As such after such a request is processed at folly layer and SSL_accept restarted, another crypto operation may return same error and thereby returning multiple WANT_ASYNC errors, which the existing folly is not handling correctly. 

The issue is that if sslSocket_->restartSSLAccept again returns WANT_ASYNC error,  asyncOperationFinishCallback_ is not be set null and further events not processed

Reply:
AsyncPipeWriter is generic and can have uses other than this use case.  I think what you want to do instead is set the closeCB_ to a no-op in your case?

Reply:
@afrind , Thanks for the review. I have removed the change so as to not break other use cases. Please review rest of the code changes.